show-line-of-sight-between-geoelements/src/main/java/com/esri/arcgismaps/sample/showlineofsightbetweengeoelements/components/SceneViewModel.kt

```diff
@@ -0,0 +1,269 @@
+import com.arcgismaps.mapping.view.GraphicsOverlay
+import com.arcgismaps.mapping.view.SurfacePlacement
+import com.esri.arcgismaps.sample.showlineofsightbetweengeoelements.R
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import java.io.File
+import kotlin.concurrent.timer
+
+class SceneViewModel(private var application: Application) : AndroidViewModel(application) {
+
+    // Create a scene and add a basemap to it
+    val scene = ArcGISScene(BasemapStyle.ArcGISTopographic)
+
+    // Create graphic overlay to hold graphics
+    val graphicsOverlay = GraphicsOverlay()
+
+    // Create an analysis overlay to hold the line of sight
+    val analysisOverlay = AnalysisOverlay()
+
+    // Keep track of target visibility status string state.
+    var targetVisibilityString by mutableStateOf("")
+
+    // Set visibility status string in the UI.
+    private fun updateTargetVisibilityString(targetVisibility: String) {
+        targetVisibilityString = targetVisibility
+    }
+
+    // Initialize z to 50 as starting point and emit its state changes
+    private val _observerHeight = MutableStateFlow(50.0)
+    val observerHeight: StateFlow<Double> = _observerHeight.asStateFlow()
+
+    // Keeps track of wayPoints
+    private var waypointsIndex = 0
+
+    // Create waypoints around a block for the taxi to drive to
+    private val wayPoints = listOf(
+        Point(-73.984513, 40.748469, SpatialReference.wgs84()),
+        Point(-73.985068, 40.747786, SpatialReference.wgs84()),
+        Point(-73.983452, 40.747091, SpatialReference.wgs84()),
+        Point(-73.982961, 40.747762, SpatialReference.wgs84()),
+    )
+
+    private val provisionPath: String by lazy {
+        application.getExternalFilesDir(null)?.path.toString() + File.separator + application.getString(
+            R.string.app_name
+        ) + File.separator
+    }
+    private val filePath = provisionPath + application.getString(R.string.dolmus_model)
+
+    // Create a symbol of a taxi using the model file
+    private val taxiSymbol = ModelSceneSymbol(
+        uri = filePath,
+        scale = 3.0F
+    ).apply {
+        anchorPosition = SceneSymbolAnchorPosition.Bottom
+    }
+
+    // Create a graphic of a taxi to be the target
+    private val taxiGraphic = Graphic(
+        geometry = wayPoints[0],
+        symbol = taxiSymbol
+    ).apply {
+        attributes["HEADING"] = 0.0
+    }
+
+    // Create a graphic near the Empire State Building to be the observer
+    private val observerGraphic = Graphic(
+        geometry = Point(
+            x = -73.9853,
+            y = 40.7484,
+            z = 50.0,
+            spatialReference = SpatialReference.wgs84()
+        ),
+        symbol = SimpleMarkerSymbol(
+            style = SimpleMarkerSymbolStyle.Circle,
+            color = Color.red,
+            size = 5f
+        )
+    )
+
+    init {
```

Comment:
I'd challenge you to try to move as much of this code out of the init block as possible. I think with relatively minor refactoring you'd be able to move out lines 135-179 at least.

Code run in an init { } block inherently adds complexity because it executes after construction. Of course use it where we have to, but where possible better to limit code done in init.